layers/+tools/google-calendar/packages.el

```diff
@@ -0,0 +1,89 @@
+;;; packages.el --- google-calendar layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2017 Sylvain Benner & Contributors
+;;
+;; Author: Markus Johansson <markus.johansson@me.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+(defconst google-calendar-packages
+  '(org-gcal
+    calfw))
+
+(defun google-calendar/init-org-gcal ()
+  "Initializes org-gcal and adds keybindings for it's exposed functions"
+  (use-package org-gcal
+    :init
+    (progn
+      (spacemacs/set-leader-keys
+        "agr" 'org-gcal-refresh-token
+        "ags" 'org-gcal-sync
+        "agf" 'org-gcal-fetch))
+    :config
+    (setq org-gcal-down-days 365)   ;; Set org-gcal to download events a year in advance
+    (add-hook 'after-init-hook 'org-gcal-fetch)
+    (add-hook 'kill-emacs-hook 'org-gcal-sync)
+    (add-hook 'org-capture-after-finalize-hook 'google-calendar/sync-cal-after-capture)
+    (run-with-idle-timer 600 t 'google-calendar/org-gcal-update)))
```

Comment:
With this, there's no way for the calendar to update while emacs is in use throughout the day. `org-gcal-fetch` is not super disruptive, so I configured my emacs to create a a much shorter idle timer every 10 minutes or so. That way it tries to sync every 10 minutes if emacs is idle for, say, 30 seconds. That way you probably won't have your typing interrupted, but it'd still update while you're using emacs normally. The code to do that is here: https://github.com/aaronjensen/spacemacs.d/blob/fc69c4a1e643ad276268a2409fc736b28ee2ae14/lisp/init-org.el#L71-L106 (probably can be cleaned up a bit, but you get the idea)